pgmpy/prediction/DynamicDMLRegressor.py

```diff
@@ -0,0 +1,851 @@
+    - Assumes positivity (overlap in treatment assignments)
+    """
+
+    def __init__(
+        self,
+        causal_graph=None,
+        model_y="auto",
+        model_t="auto",
+        cv=2,
+        n_periods=None,
+        discrete_treatment=None,
+        discrete_outcome=None,
+        random_state=None,
+    ):
+        self.causal_graph = causal_graph
+        self.model_y = model_y
+        self.model_t = model_t
+        self.cv = cv
+        self.n_periods = n_periods
+        self.discrete_treatment = discrete_treatment
+        self.discrete_outcome = discrete_outcome
+        self.random_state = random_state
+
+    def __sklearn_tags__(self):
+        """Tags for sklearn compatibility."""
+        tags = super().__sklearn_tags__()
+        tags.target_tags.required = True
+        tags.input_tags.allow_nan = False
+        tags.regressor_tags.poor_score = True
+        return tags
+
+    def _prepare_feature_df(self, X):
+        """
+        Convert input (either numpy array or dataframe) to a DataFrame and
+        validate that column names match DAG variables when causal_graph is provided.
+
+        Parameters
+        ----------
+        X : array-like or DataFrame
+            Input features.
+
+        Returns
+        -------
+        X_df : DataFrame
+            Feature DataFrame with validated columns.
+        """
+        # If no causal graph return as-is (will be validated elsewhere)
+        if self.causal_graph is None:
+            if isinstance(X, pd.DataFrame):
+                return X
+            else:
+                X_arr = np.asarray(X)
+                if X_arr.ndim == 1:
+                    raise ValueError(
+                        "Reshape your data: X must be 2D. If using a 1D array, reshape it to (n_samples, 1)."
+                    )
+                return pd.DataFrame(X_arr, columns=range(X_arr.shape[1]))
+
+        # Get required feature columns from causal graph
+        required_features = self.feature_columns_
+
+        # Convert input to DataFrame format
+        if isinstance(X, pd.DataFrame):
+            X_df = X
+        else:
+            # For numpy arrays, use range index as column names
+            X_arr = np.asarray(X)
+            if X_arr.ndim == 1:
+                raise ValueError(
+                    "Reshape your data: X must be 2D. If using a 1D array, reshape it to (n_samples, 1)."
+                )
+            X_df = pd.DataFrame(X_arr, columns=range(X_arr.shape[1]))
+
+        # Validation: column names must match DAG variables
+        missing_columns = set(required_features) - set(X_df.columns)
+        if missing_columns:
+            raise ValueError(
+                f"Missing required columns in input data: {list(missing_columns)}. "
+                f"DAG expects columns: {required_features}, but got: {list(X_df.columns)}"
+            )
+
+        return X_df[required_features]
+
+    def fit(self, X, T, y, groups):
```

Comment:
I am a bit confused about what exactly this method fits. Does this fit the whole time series or just one time slice?